### ISSUE TITLE
Cannot find FindReplaceOptions class

### ISSUE THREAD

**Issue by Golenko** [aspose-words/Aspose.Words-for-Java#37]
Hello guys,
I'm trying to execute [this example](https://github.com/aspose-words/Aspose.Words-for-Java/blob/master/Examples/src/main/java/com/aspose/words/examples/programming_documents/find_replace/ReplaceWithString.java)

The problem is that classes _FindReplaceOptions_ and _FindReplaceDirection_ are absent from _com.aspose.words_ package.

The version of the jar I use is _aspose-words-16.6.0-jdk16.jar_

Is that something that expected and example should be updated?

Thanks.

**Comment by sohail-aspose** [comment]
Hi Golenko,
Please use the latest version of [Aspose.Words for Java 16.11.0](http://www.aspose.com/downloads/words/java). 

I have tested [ReplaceWithString](https://github.com/aspose-words/Aspose.Words-for-Java/blob/master/Examples/src/main/java/com/aspose/words/examples/programming_documents/find_replace/ReplaceWithString.java) example and it is running perfectly fine. I used [this file](https://www.dropbox.com/s/6bs69msilegi87u/ReplaceWithString.doc?dl=0) as input and it produces [this file](https://www.dropbox.com/s/o6rvkbdruo7j8ri/ReplaceWithString_out.doc?dl=0) as output.

**Comment by Golenko** [comment]
Thanks for your answer!
16.6.0 is the newest version supported by the license I have.
Is there any examples for older versions?

**Comment by sohail-aspose** [comment]
Yes, three APIs are available in v16.6.0 for “Find and Replace” feature.

`replace(Pattern pattern, String replacement)`
`replace(String oldValue, String newValue, boolean isMatchCase, boolean isMatchWholeWord)`
`replace(Pattern pattern, IReplacingCallback handler, boolean isForward)`

**Example 1:**

```
Document doc = new Document(dataDir + "ReplaceWithString.doc");
doc.getRange().replace(Pattern.compile("[s|m]ad"), "bad");
doc.save(dataDir + "ReplaceWithString_out.doc");

```

**Example 2:**

```
Document doc = new Document(dataDir + "ReplaceWithString.doc");
doc.getRange().replace("sad", "bad", false, true);
doc.save(dataDir + "ReplaceWithString_out.doc");
```

**Comment by Golenko** [comment]
Thanks a lot!
